vimiv/imutils/exif.py

```diff
@@ -10,139 +10,361 @@
-                    piexif.TYPES.Ascii,
-                    piexif.TYPES.Undefined,
-                ):  # byte encoded
-                    self[keyname] = val.decode()
-                elif keytype in (
-                    piexif.TYPES.Rational,
-                    piexif.TYPES.SRational,
-                ):  # (int, int) <=> numerator, denominator
-                    self[keyname] = f"{val[0]}/{val[1]}"
+
+        return exif
+
+    def copy_exif(self, dest: str, reset_orientation: bool = True) -> None:
+        def get_copy_args():
+            """Get filetype specific copy arguments."""
+
+            file_type = imghdr.what(dest).upper()
+            args = [False, False, False, False]
+
+            if file_type in [
+                "JPEG",
+                "JPG",
+                "EXV",
+                "CR2",
+                "CRW",
+                "MRW",
+                "TIFF",
+                "TIF",
+                "WEBP",
+                "DNG",
+                "NEF",
+                "PEF",
+                "ARW",
+                "RW2",
+                "SR2",
+                "SRW",
+                "ORF",
+                "PNG",
+                "PGF",
+                "RAF",
+                "PSD",
+                "JP2",
+            ]:
+                args[0] = True
+
+            if file_type in [
+                "JPEG",
+                "JPG",
+                "EXV",
+                "CR2",
+                "MRW",
+                "TIFF",
+                "TIF",
+                "DNG",
+                "NEF",
+                "PEF",
+                "ARW",
+                "RW2",
+                "SR2",
+                "SRW",
+                "ORF",
+                "PNG",
+                "PGF",
+                "RAF",
+                "PSD",
+                "JP2",
+            ]:
+                args[1] = True
+
+            if file_type in [
+                "JPEG",
+                "JPG",
+                "EXV",
+                "CR2",
+                "MRW",
+                "TIFF",
+                "TIF",
+                "WEBP",
+                "DNG",
+                "NEF",
+                "PEF",
+                "ARW",
+                "RW2",
+                "SR2",
+                "SRW",
+                "ORF",
+                "PNG",
+                "PGF",
+                "RAF",
+                "EPS",
+                "XMP",
+                "PSD",
+                "JP2",
+            ]:
+                args[2] = True
+
+            if file_type in ["JPEG", "JPG", "EXV", "CRW", "PNG", "PGF"]:
+                args[3] = True
+
+            return tuple(args)
+
+        if reset_orientation:
+            with contextlib.suppress(KeyError):
+                self._metadata["Exif.Image.Orientation"] = ExifOrientation.Normal
+        try:
+            dest_image = pyexiv2.ImageMetadata(dest)
+            dest_image.read()
+
+            self._metadata.copy(dest_image, *get_copy_args())
+
```

Comment:
Adding
```
dest_image.write()
```
after the copying of the metadata fixes the test suite for me. Obviously the manual check for the file-types is rather broken, I will take some more time to look into this :smile:

Reply:
How could I miss something so obvious? I have expected that the pyexif based test is somehow broken. :laughing:  Thanks for the hint!

Reply:
Fixed in 9f6acd8917ae9e20932098cdac2ea3402b4fa673